internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -198,6 +200,37 @@ func (r *AgentServiceConfigReconciler) ensureAgentRoute(ctx context.Context, ins
 	return nil
 }
 
+func (r *AgentServiceConfigReconciler) ensureAgentLocalAuthSecret(ctx context.Context, instance *aiv1beta1.AgentServiceConfig) error {
+	secret, mutateFn, err := r.newAgentLocalAuthSecret(instance)
+	if err != nil {
+		conditionsv1.SetStatusCondition(&instance.Status.Conditions, conditionsv1.Condition{
```

Comment:
Should these also use the `NoHeartbeat` version?

Reply:
Yes, done.